src/bikeracks/votes.py

```diff
@@ -69,86 +69,33 @@ def submit_vote():
     # connect to db
     db = get_db()
     
+    # check if user has voted on this rack before
     vote_status = h.get_vote_status(db, rack_id, user_id)
-    if vote_status:
-        # updating an existing vote
-        
-        query = "UPDATE votes SET vote_type = ? WHERE rack_id = ? AND user_id = ?"
-        db.execute(query, (vote_type, rack_id, user_id))
-        db.commit()
-        # also update the vote count in bikeracks table
-        if vote_type == 1:
-            # update the downvote_count in bikeracks for this rack, decrement by 1
-            h.decrement_downvote_count(rack_id, db)
-            
-        # if the user downvoted
-        elif vote_type == -1:
-            # update the upvote_count in bikeracks for this rack, decrement by 1
-            h.decrement_upvote_count(rack_id, db)
-        
-    else:
-        # voting for the first time
-        
-        query = "INSERT INTO votes (rack_id, user_id, vote_type) VALUES (?, ?, ?)"
-        db.execute(query, (rack_id, user_id, vote_type))
-        db.commit()
-        
-    # increment the upvote or downvote count
-    if vote_type == 1:
-        # update the upvote_count in bikeracks for this rack, increment by 1
-        h.increment_upvote_count(rack_id, db)
+    vote_status =  h.dict_from_row(vote_status) if vote_status else None
+    # vote_status is an object, {vote_type: -1} for example or None
+    query = """ INSERT INTO votes (rack_id, user_id, vote_type)
+                VALUES (?, ?, ?)
+                ON CONFLICT(rack_id, user_id) DO UPDATE SET vote_type=?"""
+    db.execute(query, (rack_id, user_id, vote_type, vote_type))
+    db.commit()
+    
+    # if the vote is an upvote and the user has voted before
```

Comment:
It seems like you could just look at the old vote?

Assuming `vote_status[vote]` is the int representing the old vote (-1, 0, 1), consider this snippet for the six cases:
```python
if vote_type  > vote_status[vote]:
  # This is a new upvote (1, 0) or change from down to up (1, -1) or change from down to no vote (0, -1)
  delta_up = vote_type
  delta_down = vote_status[vote]
if vote_type < vote_status[vote]:
  # This is a new downvote (0, 1) or change from up to down (-1, 1) or change from up to no vote (-1, 0)
  delta_up = -vote_status[vote]
  delta_down = vote_type
h.update_vote_count(db, rack_id, delta_up, delta_down)
```

Reply:
Done.